plaso/parsers/ips_parser.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+"""Parser for IPS formatted log files. These files are used by Apple in iOS
+(15 and later) and MacOS (12 and later). Each file is a single event."""
+
+import json
+import re
+import yaml
+
+from yaml import scanner
+
+from plaso.parsers import interface
+from plaso.parsers import logger
+from plaso.parsers import manager
+
+
+class IPSFile(object):
+  """IPS log file"""
+
+  def __init__(self):
+    """Initializes an IPS log"""
+    super(IPSFile, self).__init__()
+    self.header = None
+    self.content = None
+
+  def GetContent(self, raw_content):
+    """Extracts the header from the raw content into a JSON structure.
+
+    Args:
+      raw_content (str): content of the IPS log as a string
+
+    Returns:
+      structured_content: content in a JSON structure
+    """
+    if not isinstance(raw_content, str):
+      return None
+
+    content = re.search('}\n({.+)', raw_content, re.DOTALL)
+    if content:
+      try:
+        structured_content = yaml.safe_load(content.group(1))
+        return structured_content
+
+      except (AttributeError, scanner.ScannerError):
+        return None
+
+    return None
+
+  def GetHeader(self, raw_content):
+    """Extracts the header from the raw content into a JSON structure.
+
+      Agrs:
+        raw_content (str): content of the IPS log as a string
+
+      Return:
+        structured_header: header in a JSON structure
+    """
+    if not isinstance(raw_content, str):
+      return None
+
+    header = re.search('{.+?}', raw_content)
+
+    if header:
+      try:
+        structured_header = json.loads(header.group())
+        return structured_header
+
+      except (json.decoder.JSONDecodeError, TypeError):
+        return None
+
+    return None
+
+  def Open(self, file_object):
+    """Opens an IPS log file.
+
+      Args:
+      file_object (dfvfs.FileIO): file-like object.
+
+      Raises:
+        ValueError: if the file object is missing.
+    """
+
+    if not file_object:
+      raise ValueError('Missing file object.')
+
+    raw_content = file_object.read()
+    if isinstance(raw_content, bytes):
+      try:
+        raw_content = raw_content.decode()
```

Comment:
@rick-slin what is the encoding here? UTF-8 ?

Reply:
Yes. I've only encountered UTF-8

Reply:
> Yes. I've only encountered UTF-8. For some reason, it wasn't always encoded.

in what sense? a text file always has an encoding